openpype/pipeline/colorspace.py

```diff
@@ -132,52 +197,87 @@ def get_imageio_colorspace_from_filepath(
+    """
+    if not compatibility_check():
+        # python environment is not compatible with PyOpenColorIO
+        # needs to be run in subprocess
+        result_data = get_wrapped_with_subprocess(
+            "colorspace", "get_colorspace_from_filepath",
+            config_path=config_path,
+            filepath=filepath
+        )
+        if result_data:
+            return result_data[0]
+
+    from openpype.scripts.ocio_wrapper import _get_colorspace_from_filepath
+
+    result_data = _get_colorspace_from_filepath(config_path, filepath)
+
+    if result_data:
+        return result_data[0]
+
+
 def parse_colorspace_from_filepath(
-    path, host_name, project_name,
-    config_data=None,
-    project_settings=None
+    filepath, colorspaces=None, config_path=None
 ):
     """Parse colorspace name from filepath
 
     An input path can have colorspace name used as part of name
     or as folder name.
 
+    Example:
+        >>> config_path = "path/to/config.ocio"
+        >>> colorspaces = get_ocio_config_colorspaces(config_path)
+        >>> colorspace = parse_colorspace_from_filepath(
+                "path/to/file/acescg/file.exr",
+                colorspaces=colorspaces
+            )
+        >>> print(colorspace)
+        acescg
+
     Args:
-        path (str): path string
-        host_name (str): host name
-        project_name (str): project name
-        config_data (dict, optional): config path and template in dict.
-                                      Defaults to None.
-        project_settings (dict, optional): project settings. Defaults to None.
+        filepath (str): path string
+        colorspaces (Optional[dict[str]]): list of colorspaces
+        config_path (Optional[str]): path to config.ocio file
 
     Returns:
         str: name of colorspace
     """
-    if not config_data:
-        project_settings = project_settings or get_project_settings(
-            project_name
+    if not colorspaces and not config_path:
+        raise ValueError(
+            "You need to provide `config_path` if you don't "
+            "want to provide input `colorspaces`."
         )
-        config_data = get_imageio_config(
-            project_name, host_name, project_settings)
 
-    config_path = config_data["path"]
+    colorspaces = colorspaces or get_ocio_config_colorspaces(config_path)
 
     # match file rule from path
     colorspace_name = None
-    colorspaces = get_ocio_config_colorspaces(config_path)
     for colorspace_key in colorspaces:
         # check underscored variant of colorspace name
         # since we are reformatting it in integrate.py
-        if colorspace_key.replace(" ", "_") in path:
+        if colorspace_key.replace(" ", "_") in filepath:
             colorspace_name = colorspace_key
             break
-        if colorspace_key in path:
+        if colorspace_key in filepath:
             colorspace_name = colorspace_key
             break
```

Comment:
Here's a fun trick that can be much faster potentially for many colorspaces if you need to process many files.

```python
import re


def get_colorspace_match_regex(colorspaces):
    """Return a regex which allows to search a colorspace match in a filename

        Args:
            colorspaces (list): List of colorspace names
            case_sensitive (bool, optional): Whether the match should
                be case sensitive or not. Defaults to True.

    """
    pattern = "|".join(
        # Allow to match spaces also as underscores because the
        # integrator replaces spaces with underscores in filenames
        re.escape(colorspace).replace("\ ", "[_ ]") for colorspace in 
        # Sort by longest first so the regex matches longer matches
        # over smaller matches, e.g. matching 'Output - sRGB' over 'sRGB'
        sorted(colorspaces, key=len, reverse=True)
    )
    return re.compile(pattern)


regex = get_colorspace_match_regex(colorspaces)
colorspaces = ["Utility - Raw",  'ACES - ACEScg', "Output - sRGB","acescg", "linear", "ACES - ACEScc", "sRGB"]
for filepath in ["file1.sRGB.jpg", "file2.ACES - ACEScg.exr", "/path/to/aces_testfile_acescg.exr"]:
    match = regex.search(filepath)
    colorspace = match.group(0) if match else None
    print("{: <25} --> {}".format(colorspace, filepath))
```

Reply:
By the way, [here's more details as to why this could be faster](https://stackoverflow.com/a/22945732)